### ISSUE TITLE
Use structured data in the debug buffer

### ISSUE THREAD

**Issue by xermicus** [hyperledger/solang#1048]
**Describe the bug**
Right now we just yolo everything into the debug buffer. It gets ugly and hard to read fairly quickly.

**To Reproduce**
Any contract with multiple print statements or logs many return codes of API calls (Substrate only).

**Expected behavior**
Contract execution should output something that is easy to read for humans and possible to parse by some tool. To achieve this, every write to the debug buffer could just simply be delimited by something at the start and end, or be a JSON structure. However the latter would be more complex to implement and we don't have a use case that requires it right now. It should be machine-readable, which means it must be escaped at runtime, at least the `print` statement should do this.

In summary:
Something simple, e.g. wrap every message inside quotes and append a newline or a comma would already greatly improve the situation. It will make the debug buffer much more human readable and even allow frontend tools parsing it. Printed messages need to be escaped, that will be the most work. JSON would be the cherry on top, but I'd rather have some simple improvement shipped first instead (can do fancy stuff later if it's worth it)

Favorably implement it in codegen and try to make it work for all targets.

**Comment by xermicus** [comment]
Loosely related to #971 #972

**Comment by xermicus** [comment]
Closed by #1188
